### PR TITLE
Remove TimePicker state and Moment dependency

### PR DESCRIPTION
**Pull request by pablinos**
## Description
While working on the post schedule feature, @retrofox and I noticed that there are a number of bugs with the implementation of the `DateTimePicker`. The `TimePicker` was holding its own local state, and would only update the month when the select component lost focus. 

This change updates the `TimePicker` component to:

- Not rely on local state
- Use `date-fns` rather than Moment
- Tweak the triggering of the `onChange` callback. 

## How has this been tested?

Locally using Storybook

## Types of changes

This could be a breaking change, depending on how others are using the `TimePicker` component.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/date-time/index.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ export class DateTimePicker extends Component {
 
 	render() {
 		const {
-			currentDate,
+			currentDate = new Date(),
```
Comment (pablinos) [inline packages/components/src/date-time/index.js:38]: I'm not sure this default is needed, as it's initialised elsewhere, and should probably be handled by default props instead. This is a reminder to me to remove it!

File: packages/components/src/date-time/date.js [hunk 2]
```diff
@@ -1,150 +1,53 @@
...
  */
-import ReactDatePicker from "react-datepicker";
-import "react-datepicker/dist/react-datepicker.css";
-
-/**
- * WordPress dependencies
- */
-import { Component, createRef } from '@wordpress/element';
+import set from 'date-fns/set';
+import format from 'date-fns/format';
```
Comment (ZebulanStanphill) [inline packages/components/src/date-time/date.js:5]: Correct me if I'm wrong, but couldn't this be:

```suggestion
import { format, set } from 'date-fns';
```

**Comment by pablinos** [comment]
I'm getting a problem when this is used with our version of the `DatePicker` based on React DateTimePicker. It has code to grab focus when the selected date changes. This is happening as I tab out of one of the `TimePicker`'s input fields, but strangely not when I hit enter. This is odd as it causes the same update function to be called.

File: packages/components/src/date-time/time.js [hunk 3]
```diff
@@ -3,16 +3,18 @@
  */
 import classnames from 'classnames';
 import { isInteger } from 'lodash';
-import moment from 'moment';
+import startOfMinute from 'date-fns/startOfMinute';
+import set from 'date-fns/set';
+import format from 'date-fns/format';
```
Comment (ZebulanStanphill) [inline packages/components/src/date-time/time.js:8]:
```suggestion
import { format, set, startOfMinute } from 'date-fns';
```

File: packages/components/src/date-time/time.js [hunk 4]
```diff
@@ -26,19 +28,27 @@ import TimeZone from './timezone';
...
  *
  * @param {Object}        props          Component props.
  * @param {string}        props.as       Render the component as specific element tag, defaults to "input".
  * @param {number|string} props.value    The default value of the component which will be parsed to integer.
  * @param {Function}      props.onUpdate Call back when blurred and validated.
  */
-function UpdateOnBlurAsIntegerField( { as, value, onUpdate, ...props } ) {
-	function handleBlur( event ) {
+function IntegerValidatedField( { as, value, onUpdate, ...props } ) {
+	const element = as ?? 'input';
```
Comment (ZebulanStanphill) [inline packages/components/src/date-time/time.js:43]:
```suggestion
function IntegerValidatedField( { as = 'input', value, onUpdate, ...props } ) {
	const element = as;
```

**Comment by stokesman** [comment]
I'd wager that all the problems this PR intended to address are since resolved and the last push here is going to be three years old soon. We should close this one out, no?

**Comment by pablinos** [comment]
> I'd wager that all the problems this PR intended to address are since resolved and the last push here is going to be three years old soon. We should close this one out, no?

Yes we should!

**Pull request closed without merging**